### ISSUE TITLE
Fails when used in a project that uses activesupport 5.2.0.rc1 callbacks

### ISSUE THREAD

**Issue by ball-hayden** [mvidner/ruby-dbus#71]
If ActiveSupport 5.2.0.rc1 is loaded alongside this gem, bad things happen:

```
/home/hayden/.gem/ruby/gems/activesupport-5.2.0.rc1/lib/active_support/callbacks.rb:841:in `set_callbacks': undefined method `merge' for nil:NilClass (NoMethodError)
	from /home/hayden/.gem/ruby/gems/activesupport-5.2.0.rc1/lib/active_support/callbacks.rb:812:in `block in define_callbacks'
	from /home/hayden/.gem/ruby/gems/activesupport-5.2.0.rc1/lib/active_support/callbacks.rb:809:in `each'
	from /home/hayden/.gem/ruby/gems/activesupport-5.2.0.rc1/lib/active_support/callbacks.rb:809:in `define_callbacks'
	from /home/hayden/.gem/ruby/gems/activesupport-5.2.0.rc1/lib/active_support/execution_wrapper.rb:13:in `<class:ExecutionWrapper>'
	from /home/hayden/.gem/ruby/gems/activesupport-5.2.0.rc1/lib/active_support/execution_wrapper.rb:6:in `<module:ActiveSupport>'
	from /home/hayden/.gem/ruby/gems/activesupport-5.2.0.rc1/lib/active_support/execution_wrapper.rb:5:in `<main>'
```

I suspect the same may be true of slightly less bleeding edge versions of activesupport as well, given the version that's been copied into this them is now coming up on 4 years old.

I'm thinking it's clashing with the copy of the `class_attribute` monkeypatch defined here: https://github.com/mvidner/ruby-dbus/blob/master/lib/dbus/core_ext/class/attribute.rb

I'm going to put together a fork that depends on activesupport, rather than copying the required modules, and I'll open a PR assuming there are no significant issues.

**Comment by mvidner** [comment]
Thank you for the report!

I am still considering what the best way to fix this is, but now I found that I cannot reproduce your problem. I've made a new directory with a small Gemfile:
```rb
gem "ruby-dbus"
gem "activesupport", ">= 5.2.0.rc1"
```

yet various `require` combinations all pass and do not produce your original error. Can you please give me a way to reproduce the problem?

**Comment by ball-hayden** [comment]
Thanks for looking into it!

It looks like it's not quite a simple as I first thought. The issue arose within a Rails project (don't ask why we're wanting to access DBus from there...), so there's autoloading magic making the matter a bit confusing.

The issue appears to be if ActiveSupport is loaded _around_ `dbus`, with the class attribute patch being loaded before `dbus`, and `callbacks` being loaded after `dbus` as in the following example:

```ruby
require "bundler/inline"

gemfile do
  gem "ruby-dbus", require: false
  gem "activesupport", ">= 5.2.0.rc1", require: false
  gem "pry-byebug"
end

require "active_support/core_ext/class/attribute"
require "dbus"
require "active_support/callbacks"

class Record
  include ActiveSupport::Callbacks

  define_callbacks :save
end

# => /home/hayden/.rvm/gems/ruby-2.4.1/gems/activesupport-5.2.0.rc1/lib/active_support/callbacks.rb:841:in `set_callbacks': undefined method `merge' for nil:NilClass (NoMethodError)
#   	from /home/hayden/.rvm/gems/ruby-2.4.1/gems/activesupport-5.2.0.rc1/lib/active_support/callbacks.rb:812:in `block in define_callbacks'
#   	from /home/hayden/.rvm/gems/ruby-2.4.1/gems/activesupport-5.2.0.rc1/lib/active_support/callbacks.rb:809:in `each'
#   	from /home/hayden/.rvm/gems/ruby-2.4.1/gems/activesupport-5.2.0.rc1/lib/active_support/callbacks.rb:809:in `define_callbacks'
#   	from test.rb:16:in `<class:Record>'
#   	from test.rb:13:in `<main>'
```

It feels a little contrived, but I guess that's autoloading in practice.

**Comment by ball-hayden** [comment]
For clarity - it looks like simply requiring ActiveSupport doesn't trigger this bug.
For the specific trace I originally reported, using ActiveSupport callbacks is necessary.

Interestingly, using other ActiveSupport modules that depend on `class_attribute` seems to work fine:

```ruby
require "bundler/inline"

gemfile do
  gem "ruby-dbus", require: false
  gem "activesupport", ">= 5.2.0.rc1", require: false
end

require "active_support/core_ext/class/attribute"
require "dbus"
require "active_support/number_helper/number_converter"
require "active_support/number_helper/number_to_delimited_converter"

puts ActiveSupport::NumberHelper::NumberToDelimitedConverter.new(2.5556, separator: ",").execute
# => 2,5556
```

As does the more simple case:

```ruby
require "bundler/inline"

gemfile do
  gem "ruby-dbus", require: false
  gem "activesupport", ">= 5.2.0.rc1", require: false
end

require "active_support/core_ext/class/attribute"
require "dbus"

class Test
  class_attribute :bar, instance_writer: false, default: false
end

Test.bar = true
puts Test.new.bar
# => true
```

**Comment by ball-hayden** [comment]
Ah. Of course :man_facepalming: 

To see merge being undefined, we'd need to set the `class_attribute` to default to an empty hash...

```ruby
require "bundler/inline"

gemfile do
  gem "ruby-dbus", require: false
  gem "activesupport", ">= 5.2.0.rc1", require: false
end

require "active_support/core_ext/class/attribute"
require "dbus"

class Test
  class_attribute :bar, instance_writer: false, default: {}
end

Test.bar.merge!(a: true)
puts Test.bar
# => test_3.rb:15:in `<main>': undefined method `merge!' for nil:NilClass (NoMethodError)
```

Allowing a default appears to be the breaking change in ActiveSupport that's caused this issue.

**Comment by mvidner** [comment]
Released 0.15.0 with the fix.
